### PR TITLE
[FLYW-402] Site settings redux

### PR DESCRIPTION
**Pull request by ehlertjd**
This is ready for review.

Adds [site level settings](https://flywheel-io.github.io/core/branches/site-settings-redux/swagger/index.html#/site/get_site_settings) which allow the registration of center-pays gears. 

### Review Checklist

- Tests were added to cover all code changes
- Documentation was added / updated
- Code and tests follow standards in CONTRIBUTING.md
- Extra care is taken and l)og messages are created for [Critical Data](https://github.com/flywheel-io/core/wiki/Critical-Data-Flows)

### CODE REVIEW

File: api/site/handlers/site_settings_handler.py [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+    @require_admin
+    @validators.verify_payload_exists
+    def put(self):
+        """Patch site setting values"""
+        # Validate Input
+        payload = self.request.json
+        validators.validate_data(payload, 'site-settings.json', 'input', 'PUT')
+
+        # Update/upsert
+        update_site_settings(payload)
```
Comment (roycollins) [inline api/site/handlers/site_settings_handler.py]: Could we return a 422 on InvalidInput exceptions or does it always throw just the top level 400 on exceptions?

Comment (ehlertjd) [reply api/site/handlers/site_settings_handler.py]: I think 422 is appropriate for semantic validation - will update, thanks!

File: swagger/schemas/definitions/config.json [hunk 2]
```diff
@@ -93,6 +93,31 @@
...
+        },
+
+        "site-settings": {
+            "type": "object",
+            "properties": {
+                "created": {"type": "string"},
+                "modified": {"type": "string"},
+                "center_gears": {
+                    "type": "array",
+                    "items": {"type": "string"},
```
Comment (hkethi002) [inline swagger/schemas/definitions/config.json]: We can use `{"$ref": "gear.json#/definitions/gear-name"}` to validate that the string is a valid gear name, but maybe it isn't necessary with the 422 that's raised anyways.

File: swagger/schemas/definitions/config.json [hunk 3]
```diff
@@ -93,6 +93,31 @@
...
+                }
+            },
+            "required": ["created", "modified", "center_gears"]
+        },
+        "site-settings-input": {
+            "type": "object",
+            "properties": {
+                "center_gears": {
+                    "type": "array",
+                    "items": {"type": "string"},
```
Comment (hkethi002) [inline swagger/schemas/definitions/config.json]: Same as above

File: api/site/site_settings.py [hunk 4]
```diff
@@ -0,0 +1,49 @@
+from .models import SiteSettings
+from .mappers import SiteSettingsMapper
+from ..jobs import gears
+from ..web import errors
```
Comment (kofalt) [inline api/site/site_settings.py]: What would you think about grouping these methods as static under a class?

Comment (ehlertjd) [reply api/site/site_settings.py]: I debated that - from an extensibility and testability standpoint, I think it does make more sense than functions on a module. The main goal with this submodule was to start defining a Domain API for internal consumption across package boundaries.

Comment (ehlertjd) [reply api/site/site_settings.py]: Going back and forth on this, and I think I'm going to punt for now. On one hand @ambrussimon had a good point about having to instantiate mappers via constructor. The reason I made the decision on mappers is that there may be cases where you want your mapper instance to be isolated. (e.g. you're talking to a different data source, or using an identity map)

For the interface to the domain layer a top level facade makes sense to me, and it seems that a module with "static" functions is more idiomatic than a class with static functions.

**Pull request merged**